src/phoneField/index.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { PureComponent } from 'react'
+import cc from 'classcat'
+import isEmpty from 'lodash.isempty'
+import isEqual from 'lodash.isequal'
+import SelectField from 'selectField'
+import TextField from 'textField'
+import countryTelData from 'country-telephone-data'
+import getCountryFlag from '_utils/countriecodes'
+import style from 'phoneField/style'
+
+interface PhoneFieldOnChangeParameters {
+  name: string
+  value: object
+}
+interface PhoneFieldProps  {
+  readonly id?: string
+  readonly name: string
+  readonly className?: string
+  readonly defaultPhoneCountryCodeValue?: string
+  readonly PhoneCountryCodeLabel?: string
+  readonly phoneNumberLabel?: string
+  readonly phoneNumberPlaceholder?: string
+  readonly defaultphoneNumberValue?: string
+  readonly countriesWhiteList?: string[]
+  readonly onChange?: (obj: PhoneFieldOnChangeParameters) => void
+}
+
+interface PhoneFieldState {
+  countriesDatas: object[]
+  countriesWhiteList: string[]
+  phoneNumber: string
+  phoneCountryCode: string
+  completePhoneNumber: string
+  // Make the type of [key: string]: string etc... assignable for setState()
+  [key: string]: string | string[] | object[] | number | boolean
+}
+
+const filterIso2 = (countriesList: string[]) => countryTelData.allCountries.filter(
+  country => countriesList.includes(country.iso2)
+)
+
+const mapCountriePhoneDatas = (param: allCountriesPhoneDatas[]):
+  mappedCountryPhoneDatas[] => param.map(countryDatas => ({
+    value: `+${countryDatas.dialCode}`,
+    label: `${getCountryFlag(countryDatas.iso2)} +(${countryDatas.dialCode})`,
+    ariaLabel: countryDatas.name,
+  })
+)
+
+const allCountriesPhoneDatas = mapCountriePhoneDatas(countryTelData.allCountries)
+
+export default class PhoneField extends PureComponent<PhoneFieldProps, PhoneFieldState> {
+
+  static defaultProps: Partial<PhoneFieldProps> = {
+    defaultPhoneCountryCodeValue: '',
+    defaultphoneNumberValue: '',
+    countriesWhiteList: [],
+  }
+
+  state = {
+    countriesDatas: PhoneField.countriesDatas(this.props.countriesWhiteList),
+    countriesWhiteList: this.props.countriesWhiteList,
+    phoneNumber: this.props.defaultphoneNumberValue,
+    phoneCountryCode: this.props.defaultPhoneCountryCodeValue,
+    completePhoneNumber:
+    this.props.defaultPhoneCountryCodeValue + this.props.defaultphoneNumberValue,
+  }
+
+  onChange = ({ name, value }: OnChangeParameters) => {
+    this.setState({[name]: value}, () => {
+      this.props.onChange({
```

Comment:
Won't it be better to format this to something like: { name: value } ?

Reply:
It's the case.
The onChange prop forward an object like this:
<img width="439" alt="Capture d’écran 2019-03-11 à 13 36 43" src="https://user-images.githubusercontent.com/4067977/54124494-da069a80-4402-11e9-8935-1d1d6b01eaf2.png">

Reply:
It's not the case, here you have 2 props instead of key: value. 

However, in order to retrieve both of the props (key & value), it might be better to have 2 separated components.

Reply:
It's the array in changed that isn't quite "usable" i think. 

`
changed: {
  phoneField: { ... }
}
`

Reply:
I prefer to keep this format cause it's this way we use it in forms (OneAction)